packages/next-swc/crates/next-core/src/next_app/mod.rs

```diff
@@ -208,6 +208,13 @@ impl AppPage {
         matches!(self.0.last(), Some(PageSegment::PageType(..)))
     }
 
+    pub fn is_catchall(&self) -> bool {
+        matches!(
+            self.0.last(),
```

Comment:
not necessarily correct as the completed page will have `PageSegment::PageType` as the last element (and catchall as potentially the second to last)